app/app_val_test.go

```diff
@@ -0,0 +1,89 @@
+package app
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/basecoin/modules/base"
+	wire "github.com/tendermint/go-wire"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/tendermint/tmlibs/log"
+)
+
+//-----------------------------------
+// Test cases start here
+
+func power() uint64 {
+	// % can return negative numbers, so this ensures result is positive
```

Comment:
Hmm, that's interesting because cmn.RandInt() ==> math/rand.Int() says that it only returns non-negative numbers as per https://golang.org/pkg/math/rand/#Int